### PR TITLE
Add pseudo fullscreen fallback

### PR DESCRIPTION
**Pull request by caspg**
`requestFullscreen` is not supported by iPhones ([source: mdn](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullscreen#browser_compatibility)). Because of it, `FullscreenControl` is not displayed when viewing a map on iphone.

This PR adds pseudo fullscreen mode as a fallback. Pseudo fullscreen is implemented by CSS. CSS class comes from [Leaflet plugin](https://github.com/Leaflet/Leaflet.fullscreen/blob/gh-pages/dist/leaflet.fullscreen.css#L25).

## Visuals

* Macbook - works as before, using `requestFullscren`

https://user-images.githubusercontent.com/10419439/192212231-b0273e1b-6753-4465-bc26-7b3f8dddabb0.mov

* iPhone - using pseudo fullscreen CSS

https://user-images.githubusercontent.com/10419439/192212167-7c5bbeee-d957-442a-9635-2769887d262d.mov

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Manually test the debug page (tested on MacBook and iPhone).
 - [x] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog": **feature**
 - [x] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: src/ui/control/fullscreen_control.ts [hunk 1]
```diff
@@ -29,19 +30,22 @@ class FullscreenControl implements IControl {
...
                 this._container = options.container;
             } else {
                 warnOnce('Full screen control \'container\' must be a DOM element.');
             }
         }
+
         bindAll([
             '_onClickFullscreen',
-            '_changeIcon'
+            '_onFullscreenChange'
```
Comment (HarelM) [inline src/ui/control/fullscreen_control.ts:46]: Can we remove this bindAll? Seems like a thing of the past....

**Comment by HarelM** [comment]
Thesis for taking the time to open this PR!
Can you include some visuals of how this looks?

File: src/ui/control/fullscreen_control.ts (hunk 1 above)
Comment (caspg) [reply src/ui/control/fullscreen_control.ts:46]: We need to bind `this` because those methods are passed to event listener callbacks.

Comment (HarelM) [reply src/ui/control/fullscreen_control.ts:46]: There are more modern ways to do it with typescript - using lambda or defining these methods as properties.

Comment (caspg) [reply src/ui/control/fullscreen_control.ts:46]: Nice one. Which way is preferable?

**Comment by caspg** [comment]
@HarelM Added video recordings to the PR description.

File: src/ui/control/fullscreen_control.ts (hunk 1 above)
Comment (HarelM) [reply src/ui/control/fullscreen_control.ts:46]: I personally prefer the first one. I like to see class methods look like method and not properties.

Comment (caspg) [reply src/ui/control/fullscreen_control.ts:46]: I would also prefer the first approach but I think we won't be able to remove that listener.

Comment (HarelM) [reply src/ui/control/fullscreen_control.ts:46]: Yes, you are right. Then the second one is probably the right one in this case.

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
I have no clue why this fails... :-(
Try changing order in the css or maybe remove some attributes to see if it passes the stylelint...

**Comment by caspg** [comment]
@HarelM I believe it's not the issue with CSS change but rather with the tooling. I have the same error without the change. 

I've upgraded stylelint to the latest version and there is the same error.

**Comment by HarelM** [comment]
I don't think the problem is with the css, and agree it's the tooling, but there might be a way to pass the stylelint check by changing order or something. Main branch doesn't fail this linting as far as I know so there's a need to maker this pass somehow...

**Comment by caspg** [comment]
@HarelM 🤦 The issue was that lint requires all classes to start with `maplibregl-`. I thought that lint didn't run at all but it is just failing when displaying an error message.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,10 +9,12 @@
 ### ✨ Features and improvements
 
 - *...Add new stuff here...*
+
 - Add map.getCameraTargetElevation() ([#1558](https://github.com/maplibre/maplibre-gl-js/pull/1558))
 - Add `freezeElevation` to `AnimationOptions` to allow smooth camera movement in 3D ([#1514](https://github.com/maplibre/maplibre-gl-js/pull/1514), [#1492](https://github.com/maplibre/maplibre-gl-js/issues/1492))
 - [Breaking] Remove deprecated mapboxgl css classes ([#1575](https://github.com/maplibre/maplibre-gl-js/pull/1575))
 - Add map.setStyle's transformStyle option ([#1632](https://github.com/maplibre/maplibre-gl-js/pull/1632))
+- Add pseudo (CSS) fullscreen as a fallback for iphones (#1678)
```
Comment (HarelM) [inline CHANGELOG.md:17]: Please add full link to issue here, I think you can also add yourself (github handle) if you wish, I've seen this down below and I think it's nice.

**Comment by HarelM** [comment]
Seems like everything is passing :-)
Just change the changelog comment and I'll merge this.
Thanks for taking the time to resolve this!

**Comment by caspg** [comment]
@HarelM done, thanks for reviewing

**Pull request merged**